src/modules/ledger/modules/chaincode-service-gateway/services/chaincode-service-gateway.service.ts

```diff
@@ -0,0 +1,68 @@
+import { Injectable } from '@nestjs/common';
+import { Gateway, GatewayOptions } from 'fabric-network';
+import path from 'path';
+
+import { CallContractInputDto } from '@chaincode-service-gateway/dto/contract.dto';
+import { ccp } from '@chaincode-service-gateway/helper/ccp';
+
+import { FabricWallet } from '@auth/helper/fabricWallet';
+
+const walletPath = path.join(__dirname, 'wallet');
+
+@Injectable()
+export class ChaincodeService {
+  /**
+   * Executes a contract function on the Fabric network.
+   * This function connects to the Fabric network using the provided credentials,
+   * retrieves the specified contract, and then invokes the specified function on the contract.
+   * The function can either retrieve data from the contract (evaluateTransaction) or post data to the contract (submitTransaction).
+   *
+   * @param {CallContractInputDto} callConractInputDto - An object containing the necessary credentials and contract details.
+   * @param {string} callConractInputDto.token - The identity token used to authenticate the user.
+   * @param {string} callConractInputDto.channelName - The name of the channel on the Fabric network.
+   * @param {string} callContractInputDto.contractName - The name of the contract to interact with.
+   * @param {string} callContractInputDto.functionName - The name of the function to be invoked on the contract.
+   * @param {string[]} callContractInputDto.args - The arguments to be passed to the contract function.
+   * @returns {Promise<string>} - A Promise that resolves to the result of the contract function invocation, or an error message if an error occurs.
+   */
+  async executeContractFunction(
```

Comment:
Is this function different to [callContract](https://github.com/CyFy-Blockchain/fabric-gateway/pull/20/files#diff-d49c9ecc9c318472510a149d875b5a946e529aba37f48bbab817ca90f0ea4087R153)?

Reply:
same